lib/commanded/event_store/adapters/spear/mapper.ex

```diff
@@ -140,7 +140,8 @@ defmodule Commanded.EventStore.Adapters.Spear.Mapper do
     |> Spear.Event.to_proposed_message(%{serializer_content_type => &serializer.serialize/1})
   end
 
-  def to_snapshot_data(%RecordedEvent{data: %SnapshotData{} = snapshot} = event) do
+  def to_snapshot_data(%RecordedEvent{data: %SnapshotData{} = snapshot} = event)
```

Comment:
When using the JSON instead of the TermSerializer the snapshot.data isn't a struct yet and we need to proceed to the next case which uses the JsonDecoder.

Reply:
Without this fix the snapshot.data remains just a map.